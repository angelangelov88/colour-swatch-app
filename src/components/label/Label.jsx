import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

function Label(props) {
  const {
    children,
    size,
    variant,
    className,
  } = props;

  const classes = classNames(
    'font-ibm inline-flex items-center text-xs leading-4 font-medium',
    size === 'sm' && 'px-1 py-0.5 text-xs',
    size === 'lg' && 'px-1.5 py-1 text-sm',
    variant === 'outline' && 'border-2 border-secondary-300',
    className,
  );

  return (
    <span className={classes}>
      {children}
    </span>
  );
}

Label.defaultProps = {
  children: undefined,
  size: 'lg',
  variant: 'fill',
  className: undefined,
};

Label.propTypes = {
  children: PropTypes.node,
  size: PropTypes.oneOf(['sm', 'lg']),
  variant: PropTypes.oneOf(['fill', 'outline']),
  className: PropTypes.string,
};

export default Label;
export { Label };
