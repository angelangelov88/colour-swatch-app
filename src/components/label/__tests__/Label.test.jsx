import React from 'react';
import renderer from 'react-test-renderer';
import { Label } from '../Label';

it('should render with default', () => {
  const component = renderer.create(
    <Label />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with default', () => {
  const component = renderer.create(
    <Label>Label2</Label>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with default', () => {
  const component = renderer.create(
    <Label size="lg" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with default', () => {
  const component = renderer.create(
    <Label size="sm" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with default', () => {
  const component = renderer.create(
    <Label variant="fill" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with default', () => {
  const component = renderer.create(
    <Label variant="outline" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with default', () => {
  const component = renderer.create(
    <Label className="text-red-50" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
