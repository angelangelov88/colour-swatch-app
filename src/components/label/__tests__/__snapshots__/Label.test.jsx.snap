// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with default 1`] = `
<span
  className="font-ibm inline-flex items-center text-xs leading-4 font-medium px-1.5 py-1 text-sm"
/>
`;

exports[`should render with default 2`] = `
<span
  className="font-ibm inline-flex items-center text-xs leading-4 font-medium px-1.5 py-1 text-sm"
>
  Label2
</span>
`;

exports[`should render with default 3`] = `
<span
  className="font-ibm inline-flex items-center text-xs leading-4 font-medium px-1.5 py-1 text-sm"
/>
`;

exports[`should render with default 4`] = `
<span
  className="font-ibm inline-flex items-center text-xs leading-4 font-medium px-1 py-0.5 text-xs"
/>
`;

exports[`should render with default 5`] = `
<span
  className="font-ibm inline-flex items-center text-xs leading-4 font-medium px-1.5 py-1 text-sm"
/>
`;

exports[`should render with default 6`] = `
<span
  className="font-ibm inline-flex items-center text-xs leading-4 font-medium px-1.5 py-1 text-sm border-2 border-secondary-300"
/>
`;

exports[`should render with default 7`] = `
<span
  className="font-ibm inline-flex items-center text-xs leading-4 font-medium px-1.5 py-1 text-sm text-red-50"
/>
`;
